source/core/index.ts

```diff
@@ -24,6 +24,7 @@ import urlToOptions from './utils/url-to-options';
 import optionsToUrl, {URLOptions} from './utils/options-to-url';
 import WeakableMap from './utils/weakable-map';
 import decompressResponse from './utils/decompress-response';
+import deprecationWarning from '../utils/deprecation-warning';
```

Comment:
Use [`util.deprecate`](https://nodejs.org/api/util.html#util_util_deprecate_fn_msg_code). You can pass a noop as the function.